evalml/tests/component_tests/decomposer_tests/test_decomposer.py

```diff
@@ -17,8 +29,455 @@ def test_set_time_index():
+        and y_has_time_index != "y_has_time_index"
+    ):
+        err_msg = "Too many Datetime features provided in data and provided time_index column d4t3s not present in data."
+
+    if err_msg is not None:
+        with pytest.raises(
+            ValueError,
+            match=err_msg,
+        ):
+            decomposer.fit_transform(X, y)
+    else:
+        X_t, y_t = decomposer.fit_transform(X, y)
+
+        # If the fit_transform() succeeds, assert the original X and y
+        # have unchanged indices.
+        if X_has_time_index == "X_doesnt_have_time_index":
+            assert not isinstance(X.index, pd.DatetimeIndex)
+        else:
+            assert isinstance(X.index, pd.DatetimeIndex)
+        if y_has_time_index == "y_doesnt_have_time_index":
+            assert not isinstance(y.index, pd.DatetimeIndex)
+        else:
+            assert isinstance(y.index, pd.DatetimeIndex)
+
+
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+@pytest.mark.parametrize(
+    "dataframe_has_datatime_index",
+    ["df_has_datetime_index", "df_doesnt_have_datetime_index"],
+)
+@pytest.mark.parametrize(
+    "multiple_time_features",
+    ["has_another_time_feature", "doesnt_have_another_time_feature"],
+)
+@pytest.mark.parametrize(
+    "time_index_exists",
+    ["time_index_does_exist", "time_index_does_not_exist"],
+)
+def test_decomposer_prefers_users_time_index(
+    decomposer_child_class,
+    dataframe_has_datatime_index,
+    multiple_time_features,
+    time_index_exists,
+    ts_data,
+):
+    periods = 30
+    dates_1 = pd.date_range("2020-01-01", periods=periods)
+    dates_2 = pd.date_range("2021-01-01", periods=periods)
+    dates_3 = pd.date_range("2022-01-01", periods=periods)
+    vals = np.arange(0, periods)
+    y = pd.Series(vals)
+    X = pd.DataFrame({"values": vals})
+
+    if time_index_exists == "time_index_does_exist":
+        X["dates"] = dates_1
+
+    if multiple_time_features == "has_another_time_feature":
+        X["more_dates"] = dates_2
+
+    if dataframe_has_datatime_index == "df_has_datetime_index":
+        X.set_axis(pd.DatetimeIndex(dates_3))
+
+    dec = decomposer_child_class(time_index="dates")
+
+    err_msg = None
+    if time_index_exists == "time_index_does_not_exist":
+        if multiple_time_features == "has_another_time_feature":
+            expected_values = dates_2.values
+        else:
+            err_msg = "There are no Datetime features in the feature data and neither the feature nor the target data have a DateTime index."
+    else:
+        expected_values = dates_1.values
+
+    if err_msg:
+        with pytest.raises(ValueError, match=err_msg):
+            X_t, y_t = dec.fit_transform(X, y)
+    else:
+        X_t, y_t = dec.fit_transform(X, y)
+        assert all(dec.trend.index.values == expected_values)
+
+
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+@pytest.mark.parametrize(
+    "frequency",
```

Comment:
Can we add 'MS' frequency to this list? It's the most common monthly frequency we'll encounter and was the source of our previous bug building the seasonal signal.